### ISSUE TITLE
PHP7.4 out of memory with install (homestead, symfony5)

### ISSUE THREAD

**Issue by fuzzfree** [composer/composer#8694]
Same happens with PHP7.3 - I cannot imagine I need so much memory to run a small cli symfony app. Thank you :)

My `composer.json`:

```
{
  "type": "project",
  "license": "proprietary",
  "repositories": [
    {
      "type": "vcs",
      "url": "https://github.com/fuzzfree/phpforce-soap-client-custom.git"
    }
  ],
  "require": {
    "php": "^7.4",
    "ext-bcmath": "*",
    "ext-curl": "*",
    "ext-iconv": "*",
    "ext-json": "*",
    "ext-mongodb": "*",
    "ext-simplexml": "*",
    "ext-zip": "*",
    "ext-zlib": "*",
    "elasticsearch/elasticsearch": "7.4",
    "fuzzfree/phpforce-soap-client-custom": "dev-master",
    "google/apiclient": "^2.0",
    "google/cloud-storage": "^1.14",
    "guzzlehttp/guzzle": "^6.3",
    "jbroadway/urlify": "^1.1",
    "mixpanel/mixpanel-php" : "2.*",
    "php-http/guzzle6-adapter": "^1.1",
    "sensiolabs/security-checker": "^6.0",
    "symfony/asset": "^5.0",
    "symfony/browser-kit": "^5.0",
    "symfony/console": "^5.0",
    "symfony/css-selector": "^5.0",
    "symfony/debug-pack": "*",
    "symfony/dotenv": "^5.0",
    "symfony/expression-language": "^5.0",
    "symfony/flex": "^1.0",
    "symfony/form": "^5.0",
    "symfony/framework-bundle": "^5.0",
    "symfony/maker-bundle": "^1.0",
    "symfony/monolog-bundle": "^3.1",
    "symfony/orm-pack": "^1.0",
    "symfony/process": "^5.0",
    "symfony/security-bundle": "^5.0",
    "symfony/serializer-pack": "*",
    "symfony/swiftmailer-bundle": "^3.1",
    "symfony/templating": "^5.0",
    "symfony/twig-bundle": "^5.0",
    "symfony/validator": "^5.0",
    "symfony/web-link": "^5.0",
    "symfony/yaml": "^5.0"
  },
  "require-dev": {
    "overtrue/phplint": "^1.0",
    "symfony/phpunit-bridge": "^5.0"
  },
  "config": {
    "preferred-install": {
      "*": "dist"
    },
    "sort-packages": true
  },
  "autoload": {
    "psr-4": {
      "App\\": "src/"
    }
  },
  "autoload-dev": {
    "psr-4": {
      "App\\Tests\\": "tests/"
    }
  },
  "replace": {
    "symfony/polyfill-iconv": "*",
    "symfony/polyfill-php71": "*",
    "symfony/polyfill-php70": "*",
    "symfony/polyfill-php56": "*"
  },
  "scripts": {
    "auto-scripts": {
      "cache:clear": "symfony-cmd",
      "security-checker security:check": "script"
    },
    "post-install-cmd": [
      "@auto-scripts"
    ],
    "post-update-cmd": [
      "@auto-scripts"
    ]
  },
  "conflict": {
    "symfony/symfony": "*"
  },
  "extra": {
    "symfony": {
      "id": "01C56E9RCC3A1BM12R1BC0BHZ1",
      "allow-contrib": false,
      "require": "5.0.*"
    }
  },
  "minimum-stability": "dev",
  "prefer-stable": true
}
```

Output of `composer diagnose`:

```
 WARNING
require.elasticsearch/elasticsearch : exact version constraints (7.4) should be avoided if the package follows semantic versioning
require.fuzzfree/phpforce-soap-client-custom : unbound version constraints (dev-master) should be avoided
require.symfony/debug-pack : unbound version constraints (*) should be avoided
require.symfony/serializer-pack : unbound version constraints (*) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.10.0
PHP version: 7.4.3
PHP binary path: /usr/bin/php7.4
OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019
```

When I run this command:

```
 php -d memory_limit=-1 composer.phar install
```

I get the following output:

```
1/13:       http://repo.packagist.org/p/provider-archived$95b7476cfaab2a59b34e1fbc07c1b5eaea0ffab22762f7f3baab102b0cc62822.json
    2/13:       http://repo.packagist.org/p/provider-latest$86ec1f615106d34c78091a2b129ee37221ff64e14f9ca7e96cd31a9508f6b7b8.json
    3/13:       http://repo.packagist.org/p/provider-2015$916a394c165f2c722438ca87378b2aabd74046465f01c509dfebe86d0cbbb91c.json
    4/13:       http://repo.packagist.org/p/provider-2019$309b97a3391193c6a17cf35ced0a726122a8ccd8da0ef311e3a0cf4b812ed6b3.json
    5/13:       http://repo.packagist.org/p/provider-2020-01$f879e6a660f4fd394cd266832f402b629a20094a4709fe83cf7f195228df3b9d.json
    6/13:       http://repo.packagist.org/p/provider-2019-10$719e8400065f2d085732e49fb78aa11891a710b41abcc2657d437f9a714dbe7e.json
    7/13:       http://repo.packagist.org/p/provider-2014$128d21dd4d1b844f28a7730c32fe1d8620dae62dd13fcd9baefec0af6465da7a.json
    8/13:       http://repo.packagist.org/p/provider-2019-04$7c1f3c215145c6226160243053660372d6975cee2d1b332b1fccb675e835fdab.json
    9/13:       http://repo.packagist.org/p/provider-2013$16fd94bf81f2a7432092624d3da664c71aec9d5a9f47a6779b3bf6ad4433e29d.json
    10/13:      http://repo.packagist.org/p/provider-2019-07$1b735ac736ae365a2ddd4558173eca41bc3e62aa81d6ef4ea541ba5223ac96c2.json
    11/13:      http://repo.packagist.org/p/provider-2016$e79bd7a391868401187eb927e4a5693e3f93d4023b61cd08a3abb48c26eb9659.json
    12/13:      http://repo.packagist.org/p/provider-2017$a9781fd5b37e8a8564a0cbe1335359bc69d45bfa25efe98c53d0c47bacb25217.json
    13/13:      http://repo.packagist.org/p/provider-2018$ecba4424430126e1d77b66aa7834a88109ddf19ade5d6e7c68d37175a0c71245.json
    Finished: success: 13, skipped: 0, failure: 0, total: 13
Loading composer repositories with package information                                                         Updating dependencies (including require-dev)

mmap() failed: [12] Cannot allocate memory

mmap() failed: [12] Cannot allocate memory
PHP Fatal error:  Out of memory (allocated 5142224896) (tried to allocate 4096 bytes) in phar:///home/vagrant/code/bi_dwetl/composer.phar/src/Composer/DependencyResolver/Solver.php on line 223

Fatal error: Out of memory (allocated 5142224896) (tried to allocate 4096 bytes) in phar:///home/vagrant/code/bi_dwetl/composer.phar/src/Composer/DependencyResolver/Solver.php on line 223

mmap() failed: [12] Cannot allocate memory
PHP Fatal error:  Out of memory (allocated 5142224896) (tried to allocate 4096 bytes) in phar:///home/vagrant/code/bi_dwetl/composer.phar/src/Composer/Console/Application.php on line 87

Fatal error: Out of memory (allocated 5142224896) (tried to allocate 4096 bytes) in phar:///home/vagrant/code/bi_dwetl/composer.phar/src/Composer/Console/Application.php on line 87
```

And I expected this to happen:

```
all good :)
```

**Comment by Seldaek** [comment]
Memory usage is a known issue, 2.0 will address it to some extent, so closing..

**Comment by fuzzfree** [comment]
Without xdebug the issue is solved (again it requires more than 1.5GB ram, but at least I do not have to create a 6GB vm to install packages). I remember an old notice regarding xdebug, it was not shown at this time (probably disabled releases ago). Thank you.

**Comment by Seldaek** [comment]
Unless you add COMPOSER_ALLOW_XDEBUG, Composer should automatically restart itself without Xdebug to avoid any issues.
